v2/tools/generator/pkg/names/naming.go

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package names
+
+import (
+	"strings"
+	"unicode"
+
+	"github.com/gobuffalo/flect"
+)
+
+// flectOverrides is a case-sensitive ordered list of overrides for flect pluralization
+// Earlier overrides take precedence over later ones
+var flectOverrides = []struct {
+	single string
+	plural string
+}{
+	{"Redis", "Redis"},
+	{"redis", "redis"},
+	{"FIPS", "FIPS"},
+	{"ID", "IDs"},
+	{"Id", "Ids"},
+	{"IP", "IPs"},
+	{"knownAs", "knownAs"},
+	{"KnownAs", "KnownAs"},
+	{"ssh", "sshs"},
+	{"SSH", "SSHs"},
+}
+
+// Singularize returns the singular form of the given name
+func Singularize(name string) string {
+	// work around bug in flect: https://github.com/Azure/azure-service-operator/issues/1454
```

Comment:
minor: Should this be a flect bug link?

Reply:
That particular bug has been fixed in flect, so I'll remove the link.